src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,154 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head = null;
+    private Node<T> tail = null;
+    private int size = 0;
+
+    class Node<T> {
+        private Node<T> prev;
+        private Node<T> next;
+        private T value;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.next = next;
+            this.value = value;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> tmp = tail;
+        Node<T> newNode = new Node<>(tail, value, null);
+        tail = newNode;
+        if (tmp == null) {
+            head = newNode;
+        } else {
+            tmp.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        Node<T> foundNode = getNode(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> prev = foundNode.prev;
+            Node<T> newNode = new Node<>(prev, value, foundNode);
+            foundNode.prev = newNode;
+            if (prev == null) {
+                head = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element:list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> temp = getNode(index);
+        return temp.value;
+    }
+
+    public Node<T> getNode(int index) {
```

Comment:
should this method be private? also, move this method before overrided public methods